src/main/scala/reporting/Converter.scala

```diff
@@ -329,26 +392,67 @@ object Converter {
   }
 
   def extractPredicate(chunk: BasicChunk, model: Model): HeapEntry = {
-    //this might be too simple for some cases but for prusti to tell if some
-    //variable is part of a class it should be good enough
-    //not really sure if the snap value should be added, seems to be same as one of
-    //the args in most cases.
+    // this might be too simple for some cases but for prusti to tell if some
+    // variable is part of a class it should be good enough
+    // not really sure if the snap value should be added, seems to be same as one of
+    // the args in most cases.
+    
     val argsEval = chunk.args.map(x => evaluateTerm(x, model))
-    val entry = PredHeapEntry(chunk.id.toString, argsEval)
-    entry
+    val perm: Option[Rational] = evalPerm(chunk.perm, model)
+    PredHeapEntry(chunk.id.toString, argsEval, perm)
   }
 
-  def extractSequence(
-      model: Model,
-      heap: ExtractedHeap,
-      nullRefName: String,
-      name: String,
-      elementSort: Sort,
-      encountered: Set[String]
-  ): ExtractedModelEntry = {
+  
+  def evalPerm(value: Term, model: Model): Option[Rational] = {
```

Comment:
This looks like a very simple, stripped-down interpreter for counterexample terms. Is there anything in Silicon that could already do this? If not, could this be useful for more than just evaluation permissions?